rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -206,23 +201,25 @@ private boolean allowByMinerNotBanned(Peer sender, Message message) {
      * record event if message is repeated
      */
     private boolean allowByMessageUniqueness(Peer sender, Message message) {
+        if (message.getMessageType() != MessageType.BLOCK_MESSAGE && message.getMessageType() != MessageType.TRANSACTIONS) {
+            return true;
+        }
+
         Keccak256 encodedMessage = new Keccak256(HashUtil.keccak256(message.getEncoded()));
 
-        boolean contains = receivedMessages.contains(encodedMessage);
+        synchronized (receivedPeerMessagesMap) {
+            Set<NodeID> nodeIds = receivedPeerMessagesMap.getOrDefault(encodedMessage, new HashSet<>());
```

Comment:
```suggestion
        boolean noDuplicateDetected;
        synchronized (receivedPeerMessagesMap) {
            Set<NodeID> nodeIds = receivedPeerMessagesMap.computeIfAbsent(encodedMessage, k -> new HashSet<>());
```

minor thing, but with this optimisation a new instance of `HashSet<>` would be created only if no entry found in the map (the lambda itself is stateless / doesn't capture anything, no a singleton would be used behind scenes )